common/persistence/sql/execution_tasks.go

```diff
@@ -553,6 +553,82 @@ func (m *sqlExecutionStore) RangeCompleteVisibilityTask(
 	return nil
 }
 
+func (m *sqlExecutionStore) GetTieredStorageTask(
+	request *persistence.GetTieredStorageTaskRequest,
+) (*persistence.InternalGetTieredStorageTaskResponse, error) {
+	ctx, cancel := newExecutionContext()
+	defer cancel()
+	rows, err := m.Db.SelectFromTieredStorageTasks(ctx, sqlplugin.TieredStorageTasksFilter{
+		ShardID: request.ShardID,
+		TaskID:  request.TaskID,
+	})
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return nil, serviceerror.NewNotFound(fmt.Sprintf("GetTieredStorageTask operation failed. Task with ID %v not found. Error: %v", request.TaskID, err))
+		}
+		return nil, serviceerror.NewUnavailable(fmt.Sprintf("GetTieredStorageTask operation failed. Failed to get record. TaskId: %v. Error: %v", request.TaskID, err))
+	}
+
+	if len(rows) == 0 {
+		return nil, serviceerror.NewUnavailable(fmt.Sprintf("GetTieredStorageTask operation failed. Failed to get record. TaskId: %v", request.TaskID))
```

Comment:
seems that this should return not found error, not unavailable
can you also make similar changes to other get task APIs?

Reply:
Done.